testing-modules/mocks/src/test/java/com/baeldung/jimfs/FileRepositoryUnitTest.java

```diff
@@ -0,0 +1,74 @@
+package com.baeldung.jimfs;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class FileRepositoryUnitTest implements FileTestProvider {
+
+    private final FileRepository fileRepository = new FileRepository();
+
+    private static Stream<Arguments> provideFileSystem() {
```

Comment:
Can we format this the same as FileManipulationUnitTest? Actually rather than duplicate this provider in the two tests, could we maybe move it into the interface FileTestProvider and then reference it with a fully qualified name in our tests. e.g) @MethodSource("com.baeldung.jimfs.FileTestProvider#provideFileSystem")